src/main/java/nextstep/security/oauth2/authentication/OAuth2AuthorizationRequestRedirectFilter.java

```diff
@@ -0,0 +1,60 @@
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.security.authentication.AuthenticationException;
+import nextstep.security.oauth2.exchange.AuthorizationRequestRepository;
+import nextstep.security.oauth2.exchange.OAuth2AuthorizationRequest;
+import nextstep.security.oauth2.exchange.OAuth2AuthorizationRequestResolver;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+
+public class OAuth2AuthorizationRequestRedirectFilter extends OncePerRequestFilter {
+
+    private final OAuth2AuthorizationRequestResolver authorizationRequestResolver;
+    private final AuthorizationRequestRepository authorizationRequestRepository;
+
+    public OAuth2AuthorizationRequestRedirectFilter(OAuth2AuthorizationRequestResolver authorizationRequestResolver,
+                                                    AuthorizationRequestRepository authorizationRequestRepository) {
+
+        this.authorizationRequestResolver = authorizationRequestResolver;
+        this.authorizationRequestRepository = authorizationRequestRepository;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+
+        try {
+            OAuth2AuthorizationRequest authorizationRequest = this.authorizationRequestResolver.resolve(request);
+            if (authorizationRequest != null) {
+                this.sendRedirectForAuthorization(request, response, authorizationRequest);
+                return;
+            }
+        } catch (Exception e) {
+            throw new AuthenticationException("exception ... %s".formatted(e.getMessage()));
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private void sendRedirectForAuthorization(HttpServletRequest request, HttpServletResponse response,
+                                              OAuth2AuthorizationRequest authorizationRequest) throws IOException {
+
+        this.authorizationRequestRepository.saveAuthorizationRequest(authorizationRequest, request, response);
+
+        UriComponents redirectUrl = UriComponentsBuilder.fromHttpUrl(authorizationRequest.getAuthorizationRequestUri())
+                                                        .queryParam("client_id", authorizationRequest.getClientId())
+                                                        .queryParam("response_type", "code")
+                                                        .queryParam("scope", authorizationRequest.getScopes())
+                                                        .queryParam("redirect_uri", authorizationRequest.getRedirectUri())
+                                                        .build();
+
+        response.sendRedirect(redirectUrl.toUriString());
```

Comment:
OAuth2AuthorizationRequestResolver에서 OAuth2AuthorizationRequest를 생성할 때, authorizationRequestUri에 필수적인 쿼리 파라미터(client_id, response_type, scope, redirect_uri 등)를 미리 포함할 수도 있을 것 같은데, 어떻게 생각하시나요? 😄 

```suggestion
        response.sendRedirect(authorizationRequest.getAuthorizationRequestUri());
```

Reply:
이부분은 Redirect Filter가 모든 Redirect Url 스펙을 알게 되는 것 같아서, 생성 책임을 Resolver에게 맡기는 것이 좋아보이네요 !! 👍